src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -199,10 +205,12 @@ private bool IsServerValidForCluster(ClusterType clusterType, ClusterConnectionM
                 case ClusterType.Unknown:
                     switch (connectionMode)
                     {
+#pragma warning disable 618
                         case ClusterConnectionMode.Automatic:
+#pragma warning restore 618
                             if (serverType == ServerType.Standalone)
                             {
-                                return Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv; // Standalone is only valid in MultiServerCluster when using MongoDBPlusSrv scheme
+                                return _servers.Count == 1 || Settings.Scheme == ConnectionStringScheme.MongoDBPlusSrv;
```

Comment:
this change is still required to pass this test: https://github.com/mongodb/specifications/commit/e56f5eceed7729f8b9b43a4a1f76c7e5840db49f#diff-1e43c16a36354420446f6a2f3e19b10eR2